src/main/java/org/example/persistence/PersistenceProgram.java

```diff
@@ -1,54 +1,422 @@
+            stmt.setLong(3,programEntity.getPrograminformationId());
+            stmt.executeUpdate();
+
+            return true;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+
+    }
+
+    public ArrayList<CreditEntity> getCreditUser(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT credit.program, \"user\".id, \"user\".title, \"user\"" +
+                    ".firstname, \"user\".middlename ,\"user\".lastname,\"user\".email from credit inner join " +
+                    "\"user\" on credit.\"user\" = \"user\".id where credit.program = ? ");
+            statement.setLong(1,programEntity.getId());
+            ResultSet resultSet = statement.executeQuery();
+            ArrayList<CreditEntity> creditList = new ArrayList<>();
+            while (resultSet.next()){
+               CreditEntity credit = new CreditEntity(programEntity.getId(),createUserEntityFromResultSet(resultSet));
+               creditList.add(credit);
+            }
+            return creditList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public ArrayList<UserEntity> getProducerforProgram(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement("select \"user\".id, \"user\".title,firstname, middlename, lastname, " +
+                    "email from programproducer inner join \"user\" on programproducer.producer_id = \"user\".id " +
+                    "where programproducer.program_id = ?");
+            statement.setLong(1,programEntity.getId());
+            ResultSet resultSet = statement.executeQuery();
+            ArrayList<UserEntity> producerList = new ArrayList<>();
+            while (resultSet.next()){
+            UserEntity producer = createUserEntityFromResultSet(resultSet);
+            producerList.add(producer);
+            }
+            return producerList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+
+
+    }
+
+    public CompanyEntity getCompanyProgram(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT company.name , company.id from company" +
+                    " inner join companyprogram  on company.id = companyprogram.company where program = ?");
+            statement.setLong(1,programEntity.getId());
+            ResultSet resultSet = statement.executeQuery();
+
+            if (!resultSet.next()) {
+                return null;
+            }
+
+            CompanyEntity company = new CompanyEntity(resultSet.getLong("id"),resultSet.getString("name"));
+
+
+            return company;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
```

Comment:
Hvorfor beder du om et helt object, når du kun bruger id'et for det? hvad er tanken bag det?